core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -108,6 +108,22 @@ public NativeImageBuildItem build(NativeConfig nativeConfig, NativeImageSourceJa
                 nativeImage.add("--publish=" + DEBUG_BUILD_PROCESS_PORT + ":" + DEBUG_BUILD_PROCESS_PORT);
             }
             Collections.addAll(nativeImage, "--rm", nativeConfig.builderImage);
+
+            if ("docker".equals(containerRuntime) || "podman".equals(containerRuntime)) {
+                // we pull the docker image in order to give users an indication of which step the process is at
+                // it's not strictly necessary we do this, however if we don't the subsequent version command
+                // will appear to block and no output will be shown
+                log.info("Pulling image " + nativeConfig.builderImage);
+                try {
+                    Process pullProcess = new ProcessBuilder(Arrays.asList(containerRuntime, "pull", nativeConfig.builderImage))
+                            .inheritIO()
+                            .start();
+                    pullProcess.waitFor();
```

Comment:
Better call process.destroy() in a finally block just in case

Reply:
Good idea

Reply:
Done

Reply:
Maybe have a look at that change by Stuart: https://github.com/quarkusio/quarkus/pull/6031 .

Reply:
Hm... In this case we aren't really reading anything from the process output, we just want to wait for it, so I don't think the change is needed, right?